src/Microsoft.VisualStudio.Services.Agent/ConfigurationStore.cs

```diff
@@ -46,6 +46,12 @@ public sealed class AgentSettings
 
         [DataMember(EmitDefaultValue = false)]
         public int DeploymentGroupId { get; set; }
+
+        [DataMember(EmitDefaultValue = false)]
+        public string ProjectId { get; set; }
```

Comment:
no this agent will not work with old sever. Setting file will only have projectId.
We need to support projectName only for old agents which will get upgrade with new bits and need to be unconfigured.